### PR TITLE
[risk=low][no ticket] Partial functional refactoring of AdminInstitutionEdit

### PR DESCRIPTION
**Pull request by jmthibault79**
Extract a TierConfig functional component from AdminInstitutionEdit, as well as a few smaller ones.

I have further plans for refactoring this code, mainly involving error state.  But this is a good checkpoint and a decent size for independent review.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/admin/admin-institution-edit.tsx [hunk 1]
```diff
@@ -100,6 +102,169 @@ enum InstitutionMode {
...
+  const {tierConfig, onChange} = props;
+  return <Dropdown style={{width: '16rem'}}
+                   data-test-id={`${tierConfig.accessTierShortName}-agreement-dropdown`}
+                   placeholder='Select type'
+                   options={MembershipRequirements}
+                   value={tierConfig.membershipRequirement}
+                   onChange={(v) => onChange(v.value)}/>;
+};
+
+const AddressTextArea = (props: {accessTierShortName: string, emailAddresses: string[], onBlur: Function, onChange: (string) => void}) => {
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.tsx:141]: I will probably combine these 2 in the next iteration

File: ui/src/app/pages/admin/admin-institution-edit.spec.tsx [hunk 2]
```diff
@@ -51,54 +52,54 @@ describe('AdminInstitutionEditSpec', () => {
...
   });
 
-  it('should hide/show controlled card when controlled tier disabled/enabled', async() => {
+  it('should hide/show CT card details when controlled tier disabled/enabled', async() => {
     const wrapper = component();
     await waitOneTickAndUpdate(wrapper);
     expect(wrapper).toBeTruthy();
 
-    const ctEnabledToggle = wrapper.find('[data-test-id="ct-enabled-switch"]').first().instance() as InputSwitch;
+    const ctEnabledToggle = wrapper.find('[data-test-id="controlled-enabled-switch"]').first().instance() as InputSwitch;
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.spec.tsx:60]: changes in this file are almost entirely like this.  Regularizing the naming of these helped me make the components more generic.

File: ui/src/app/pages/admin/admin-institution-edit.tsx [hunk 3]
```diff
@@ -536,133 +694,37 @@ export const AdminInstitutionEdit = fp.flow(withNavigation, withRouter)(class ex
...
-                  </p>
-                </FlexColumn>}
-              </div>}
-            </FlexColumn>
-          </FlexRow>
-        </FlexRow>
+          <TierConfig
+              institution={institution}
+              accessTierShortName={AccessTierShortNames.Registered}
+              setEraRequired={(value) => this.setRtRequireEra(value)}
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.tsx:700]: next iteration idea: consolidate the `set` functions

File: ui/src/app/pages/admin/admin-institution-edit.tsx [hunk 4]
```diff
@@ -536,133 +694,37 @@ export const AdminInstitutionEdit = fp.flow(withNavigation, withRouter)(class ex
...
+              setTierRequirement={(requirement) => this.setRegisteredTierRequirement(requirement)}
+              validateTierAddresses={() => this.validateRtEmailAddresses()}
+              setTierAddresses={(addrs) => this.setRegisteredTierEmails(addrs)}
+              validateTierDomains={() => this.validateRtEmailDomains()}
+              setTierDomains={(domains) => this.setRegisteredTierDomains(domains)}
+              invalidAddress={this.state.invalidRtEmailAddress}
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.tsx:706]: next iteration idea: consolidate the error messages

File: ui/src/app/pages/admin/admin-institution-edit.tsx [hunk 5]
```diff
@@ -1,3 +1,10 @@
+import * as fp from 'lodash/fp';
+import {Dropdown} from 'primereact/dropdown';
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-institution-edit.tsx:2]: not adding a Dropdown, just a new ordering of imports :)

File: ui/src/app/pages/admin/admin-institution-edit.tsx [hunk 6]
```diff
@@ -100,6 +102,169 @@ enum InstitutionMode {
...
+              onChange={setTierAddresses}/>
+            {invalidAddress && <div data-test-id={`${accessTierShortName}-email-address-error`} style={{color: colors.danger}}>
+              {invalidAddressMsg}
+            </div>}
+            <p style={{color: colors.primary, fontSize: '12px', lineHeight: '18px'}}>
+              Enter one email address per line.  <br/>
+            </p>
+          </FlexColumn>}
+          {tierConfig.membershipRequirement === InstitutionMembershipRequirement.DOMAINS
+          && <FlexColumn data-test-id={`${accessTierShortName}-email-domain`} style={{width: '16rem'}}>
```
Comment (calbach) [inline ui/src/app/pages/admin/admin-institution-edit.tsx:218]: nit: would make && wrapping and indendation consistent with above

File: ui/src/app/pages/admin/admin-institution-edit.tsx [hunk 7]
```diff
@@ -100,6 +102,169 @@ enum InstitutionMode {
...
+    invalidAddress, invalidAddressMsg, invalidDomain, invalidDomainMsg} = props;
+
+  const tierConfig = getTierConfig(institution, accessTierShortName);
+  const {emailAddresses, emailDomains} = tierConfig;
+
+  return <FlexRow style={styles.tierConfigContainer}>
+    <FlexColumn>
+      {accessTierShortName === AccessTierShortNames.Registered
+          ? <RegisteredTierBadge style={{marginTop: '0.6rem', marginLeft: '0.6rem'}}/>
+          : <ControlledTierBadge style={{marginTop: '0.6rem', marginLeft: '0.6rem'}}/>}
```
Comment (calbach) [inline ui/src/app/pages/admin/admin-institution-edit.tsx:185]: nit: I feel this is the one spot that slightly breaks separation of concerns; maybe make the badge content a prop or else change this to be a switchCase? Issue being that there's an implicit assumption that there's only two tiers, i.e. if ! registered, we assume controlled.

**Review by calbach — APPROVED** [review]

**Pull request merged**